### PR TITLE
AG-1409 reset default sort bug

### PR DESCRIPTION
**Pull request by sagely1**
Bug in sorting direction when switching categories.

### CODE REVIEW

File: tests/gene-comparison-tool.spec.ts [hunk 1]
```diff
@@ -27,4 +27,46 @@ test.describe('specific viewport block', () => {
...
     await expect(dropdown).toHaveText('Targeted Selected Reaction Monitoring (SRM)');
   });
+
+  test('switching from RNA to Protein with RNA-specific column ordering reverts back to Risk Score descending', async ({ page }) => {
+    // set category for Protein - Differential Expression
+    await page.goto('/genes/comparison?sortField=FP&sortOrder=1');
+
+    // wait for page to load (i.e. spinner to disappear)
+    await expect(page.locator('div:nth-child(4) > div > .spinner'))
+      .not.toBeVisible({ timeout: 150000});
```
Comment (hallieswan) [inline tests/gene-comparison-tool.spec.ts]: Consider using the newly added helper method here? 

```suggestion
    await waitForSpinnerNotVisible(page, 150000);
```

File: tests/gene-comparison-tool.spec.ts [hunk 2]
```diff
@@ -27,4 +27,46 @@ test.describe('specific viewport block', () => {
...
+    // Gene Comparison Overview tutorial modal
+    const tutorialModal = page.getByText('Gene Comparison Overview');
+    await expect(tutorialModal).toBeVisible({ timeout: 10000});
+
+    // close the Gene Comparison Overview tutorial modal
+    const closeButton = page.locator('.p-dialog-header-close-icon');
+    await closeButton.click();
+
+    // Gene Comparison Overview tutorial modal
+    await expect(tutorialModal).not.toBeVisible({ timeout: 10000});
```
Comment (hallieswan) [inline tests/gene-comparison-tool.spec.ts]: Consider using another newly added helper method here as well?

```suggestion
    await closeGctHelpDialog(page);
```

File: tests/gene-comparison-tool.spec.ts [hunk 3]
```diff
@@ -27,4 +27,46 @@ test.describe('specific viewport block', () => {
...
+    await FPColumn.click();
+    // second click sorts ascending
+    await FPColumn.click();
+
+    // expect url to be correct
+    expect(page.url()).toBe('http://localhost:8080/genes/comparison?sortField=FP&sortOrder=1');
+
+    // change category to Protein
+    await page.locator('p-dropdown').filter({ hasText: 'RNA - Differential Expression' }).getByLabel('dropdown trigger').click();
+    await page.getByText('Protein - Differential Expression').click();
```
Comment (hallieswan) [inline tests/gene-comparison-tool.spec.ts]: Could also use a newly added helper method here -- 

```suggestion
    await changeGctCategory(page, GCT_CATEGORIES.RNA, GCT_CATEGORIES.PROTEIN);
```

File: tests/gene-comparison-tool.spec.ts [hunk 4]
```diff
@@ -27,4 +27,46 @@ test.describe('specific viewport block', () => {
...
+
+    // sort by FP ascending
+    const FPColumn = page.getByText('FP');
+    // first click sorts descending
+    await FPColumn.click();
+    // second click sorts ascending
+    await FPColumn.click();
+
+    // expect url to be correct
+    expect(page.url()).toBe('http://localhost:8080/genes/comparison?sortField=FP&sortOrder=1');
```
Comment (hallieswan) [inline tests/gene-comparison-tool.spec.ts]: Could use newly added constants -- 

```suggestion
    expect(page.url()).toBe(`${baseURL}${URL_GCT}?sortField=FP&sortOrder=1`);
```

File: tests/gene-comparison-tool.spec.ts [hunk 5]
```diff
@@ -27,4 +27,46 @@ test.describe('specific viewport block', () => {
...
+
+    // change category to Protein
+    await page.locator('p-dropdown').filter({ hasText: 'RNA - Differential Expression' }).getByLabel('dropdown trigger').click();
+    await page.getByText('Protein - Differential Expression').click();
+
+    // expect url to be correct
+    expect(page.url()).toBe('http://localhost:8080/genes/comparison?category=Protein+-+Differential+Expression');
```
Comment (hallieswan) [inline tests/gene-comparison-tool.spec.ts]: Could use constants here as well -- 

```suggestion
    // expect url to be correct
    expect(page.url()).toBe(`${baseURL}${URL_GCT_PROTEIN}`);
```

File: tests/gene-comparison-tool.spec.ts [hunk 6]
```diff
@@ -27,4 +27,46 @@ test.describe('specific viewport block', () => {
     const dropdown = page.locator('#subCategory');
     await expect(dropdown).toHaveText('Targeted Selected Reaction Monitoring (SRM)');
   });
+
+  test('switching from RNA to Protein with RNA-specific column ordering reverts back to Risk Score descending', async ({ page }) => {
+    // set category for Protein - Differential Expression
+    await page.goto('/genes/comparison?sortField=FP&sortOrder=1');
```
Comment (hallieswan) [inline tests/gene-comparison-tool.spec.ts]: Could use a newly added constant here -- 

```suggestion
    // set category for Protein - Differential Expression
    await page.goto(`${URL_GCT}?sortField=FP&sortOrder=1`);
```

**Review by hallieswan — APPROVED** [review]
Looks good to me! A couple suggestions for how to use the constants and helper methods added in https://github.com/Sage-Bionetworks/Agora/pull/1304, if you'd like

File: tests/gene-comparison-tool.spec.ts (hunk 6 above)
Comment (sagely1) [reply tests/gene-comparison-tool.spec.ts]: good idea!

File: tests/gene-comparison-tool.spec.ts (hunk 4 above)
Comment (sagely1) [reply tests/gene-comparison-tool.spec.ts]: done

File: tests/gene-comparison-tool.spec.ts (hunk 3 above)
Comment (sagely1) [reply tests/gene-comparison-tool.spec.ts]: using new helper method

File: tests/gene-comparison-tool.spec.ts (hunk 1 above)
Comment (sagely1) [reply tests/gene-comparison-tool.spec.ts]: using new helper method

File: tests/gene-comparison-tool.spec.ts (hunk 2 above)
Comment (sagely1) [reply tests/gene-comparison-tool.spec.ts]: using new helper method

**Pull request merged**